server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -70,51 +71,39 @@ public void executeOrFail(DependencyCarrier dependencies,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) throws Exception {
-        var addColumnRequest = createRequest(alterTable, dependencies.nodeContext(), plannerContext,
-                                                              params, subQueryResults, dependencies.fulltextAnalyzerResolver());
-
-        dependencies.alterTableOperation().executeAlterTableAddColumn(addColumnRequest)
-            .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
-    }
-
-    public static AddColumnRequest createRequest(AnalyzedAlterTableAddColumn alterTable,
-                                                 NodeContext nodeContext,
-                                                 PlannerContext plannerContext,
-                                                 Row params,
-                                                 SubQueryResults subQueryResults,
-                                                 FulltextAnalyzerResolver fulltextAnalyzerResolver) {
         var tableElements = validate(
             alterTable,
             plannerContext.transactionContext(),
-            nodeContext,
+            dependencies.nodeContext(),
             params,
             subQueryResults,
-            fulltextAnalyzerResolver
+            dependencies.fulltextAnalyzerResolver()
         );
+        var addColumnRequest = createRequest(tableElements, alterTable.tableInfo().ident());
 
-        // We can add multiple object columns via ALTER TABLE ADD COLUMN.
-        // Those columns can have overlapping paths, for example we can add columns o['a']['b'] and o['a']['c'].
-        // For every added column AnalyzedColumnDefinition provides not only leaf but also path to the root.
-        // For o['a']['b'] and o['a']['c'] we can end up having Ref(Ident(o)) and Ref(Ident(a)) twice.
-
-        // We need a Map to compare References by FQN.
-        // Regular Set cannot be used as it would use Reference.position along with other fields when calling equals() and
-        // position is resolved to -1 and -2 for all parts of the path in the case above so overlapping parts will be "different".
```

Comment:
Moved this to AnalyzedTableElelemnts  javadocs as we now reuse this on table creation